debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -328,6 +329,23 @@ protected void tryStartingSnapshot(RelationalSnapshotContext<P, O> snapshotConte
         }
     }
 
+    /**
+     * For the given table Sets source.ts_ms value for the snapshot data!
+     */
+    protected Instant getSourceCurrentTimestamp(Table table) {
+        try {
+            // to get TS from the database. To avoid any mismatch between jvm time and DB time.
+            return jdbcConnection.getCurrentTimestamp();
+        }
+        catch (SQLException e) {
+            throw new ConnectException("Getting source timestamp failed", e);
+        }
+        // other options using JVM timestamp
+        // final Instant sourceTableSnapshotTimestamp = getClock().currentTimeAsInstant();
+        // final Instant sourceTableSnapshotTimestamp0 = Instant.ofEpochSecond(0);
+        // return sourceTableSnapshotTimestamp;
```

Comment:
other options, simpler when using jvm based TS. commented out its just for displaying the alternatives

Reply:
It is easier, but let's consider a hypothetical case.  

Let's say you need to snapshot a large data set, and it will take several hours to complete.  If you use the JVM approach, the issue is that the timestamp advances for each table, but the point in time for which the flashback query is based is static across all tables.  

I understand we have this problem even now, but if we're going to go through the exercise to change the behavior, I think it wise to do it so we are as accurate as possible.

I just don't see using the current clock time any better than what we are doing now.  I'd rather we use `0` or we go through the motions to resolve that value per connector based on whatever logic each connector requires.  For MySQL that may be `SELECT CURRENT_TIMESTAMP` but for Oracle that may be `SELECT SCN_TO_TIMESTAMP(?) FROM DUAL`, etc.

Reply:
Tested changes with Mysql and Postgresql its possible to get TS reliably because snapshot process is done using setAutoCommit=false

- MySQL: somehow with auto commit OFF CURRENT_TIMESTAMP is still increasing. i believe only single table select is transactional.
- PostgreSQL: Full snapshot process is transactional. and CURRENT_TIMESTAMP returns static transaction TS for all tables. 
- Oracle:  Full snapshot process is transactional and getting TS from SCN makes it consistent.
- SQL server:  according [to documentation](https://docs.microsoft.com/en-us/sql/connect/jdbc/reference/setautocommit-method-sqlserverconnection?view=sql-server-ver16#remarks) it should be consistent similar behavior like Postgresql.
- Mongodb:  its seems already consistent no change necessary.

so only Mysql has special case for other databases CURRENT_TIMESTAMP is same as transaction timestamp.